unit_tests/test_redis_aggregator.py

```diff
@@ -0,0 +1,238 @@
+# pylint: disable=missing-function-docstring, protected-access
+import json
+import uuid
+from unittest.mock import patch, PropertyMock, MagicMock
+import pytest
+
+from gsy_e_sdk.redis_aggregator import RedisAggregator, RedisAggregatorAPIException
+
+TEST_AGGREGATOR_NAME = "TestAgg"
+
+TEST_AREA_NAME = "TestArea"
+TEST_AREA_MAPPING = {TEST_AREA_NAME: [str(uuid.uuid4())]}
+
+TEST_TRANSACTION_ID = str(uuid.uuid4())
+TEST_AGGREGATOR_UUID = str(uuid.uuid4())
+
+TEST_DEVICE_UUID_1 = str(uuid.uuid4())
+TEST_DEVICE_UUID_2 = str(uuid.uuid4())
+TEST_BATCH_COMMAND = "TEST_BATCH_COMMAND"
+
+TEST_BATCH_COMMAND_DICT = {
+    TEST_DEVICE_UUID_1: [TEST_BATCH_COMMAND],
+    TEST_DEVICE_UUID_2: [TEST_BATCH_COMMAND]
+}
+
+TEST_TRANSACTION_ID_RESPONSE = "TEST_TRANSACTION_ID_RESPONSE"
+TEST_TRANSACTION_ID_BUFFER = {
+    TEST_TRANSACTION_ID: {"transaction_id": TEST_TRANSACTION_ID,
+                          "response": TEST_TRANSACTION_ID_RESPONSE}
+}
+
+TEST_RESPONSE = "TEST_RESPONSE"
+TEST_TRANSACTION_ID_RESPONSE_BUFFER = {TEST_TRANSACTION_ID: TEST_RESPONSE}
+
+
+@pytest.fixture(autouse=True)
+def fixture_mock_connections(mocker):
+    mocker.patch("gsy_e_sdk.redis_aggregator.StrictRedis")
+    mocker.patch("gsy_e_sdk.redis_aggregator.ThreadPoolExecutor")
+
+
+@pytest.fixture(name="mock_grid_fee_calculation")
+def fixture_mock_grid_fee_calculation(mocker):
+    """Mock GridFeeCalculation class."""
+    mocker.patch("gsy_e_sdk.redis_aggregator.GridFeeCalculation")
+
+
+@pytest.fixture(name="mock_transaction_id_and_timeout_blocking")
+def fixture_mock_transaction_id_and_timeout_blocking(mocker):
+    mocker.patch("gsy_e_sdk.redis_aggregator.uuid.uuid4",
+                 return_value=TEST_TRANSACTION_ID)
+    mocker.patch("gsy_e_sdk.redis_aggregator.wait_until_timeout_blocking")
+
+
+@pytest.fixture(name="mock_client_command_buffer_attributes")
+def fixture_mock_client_command_buffer(mocker):
+    mocker.patch("gsy_e_sdk.redis_aggregator.ClientCommandBuffer.buffer_length",
+                 new_callable=PropertyMock, return_value=3)
+    mocker.patch("gsy_e_sdk.redis_aggregator.ClientCommandBuffer.execute_batch",
+                 return_value=TEST_BATCH_COMMAND_DICT)
+
+
+@pytest.fixture(name="aggregator")
+def fixture_aggregator(mocker):
+    mocker.patch("gsy_e_sdk.redis_aggregator.RedisAggregator._create_aggregator")
```

Comment:
I have the feeling that this is not needed. It is a private method, which shouldn't be mocked in general, (that's why it caught my attention). Also, we're already mocking `StrictRedis`, so `_create_aggregator` shouldn't do much I suppose. What you probably wanted to avoid is the waiting time on `wait_until_timeout_blocking()`, but we could probably handle this issue in a different way (and without a global fixture).

Moreover: I'm having the feeling that we could benefit from using a fake redis server for our unit tests. We do it for S3 (with fakes3) so I guess there wouldn't be anything wrong.